app/views/admin/assets/index.html.erb

```diff
@@ -39,9 +39,10 @@
               <div>N comments / N listens</div>
             </div>
             <div class="button_column">
-              <%= button_to "Mark As Spam", spam_admin_asset_path(asset.id), method: :put %>
-              <%= button_to "Spam All for IP", mark_group_as_spam_admin_assets_path, method: :put, params: { mark_spam_by: { user_id: asset.user_id } } %>
-              <%= button_to "Delete User", admin_user_path(asset.user_id), data: { confirm: 'This will delete user forevAr. Are you sure?'}, method: :delete %>
+              <%= button_to "Mark As Spam", spam_admin_asset_path(asset.id), method: :put, remote: true, data: { target: 'admin--asset.spamButton', action: 'click->admin--asset#showUnspamButton'} %>
+              <%= button_to "Mark As Ham", unspam_admin_asset_path(asset.id), method: :put, remote: true, data: { target: 'admin--asset.unspamButton', action: 'click->admin--asset#showSpamButton'} %>
+              <%#= button_to "Spam All for IP", mark_group_as_spam_admin_assets_path, method: :put, params: { mark_spam_by: { user_id: asset.user_id } } %>
+              <%= button_to "Delete User", delete_admin_user_path(asset.user&.login), data: { confirm: 'This will delete user forevAr. Are you sure?'}, method: :put, remote: true, data: { target: 'admin--asset.deleteUserButton' } %>
```

Comment:
We probably don't want to delete user here, but track. Also, we don't need the confirmation if it's soft delete, which.... I guess it isn't, but should be? Hmmm.....

Reply:
I think the original idea was that you could delete a user that you see is creating nothing but spam tracks. Deleting user will soft delete assets. Let's comment this out and create a new ticket to add the ability to delete assets? Just so that we can push this PR out.